### ISSUE TITLE
Add every() method?

### ISSUE THREAD

**Issue by lawsie** [lawsie/guizero#55]
As people are often confused by the use of `.after()` I was wondering about how to simplify it. I wrote this code whcih 1) exposes the `after()` method, but converts the time to seconds, and 2) creates a new method called `every()` which you can attach to a widget. The purpose is to automatically register a callback to a function every n seconds, so that people can just write the function and not have to worry about putting the `after()` line inside it to call it again. I think this would also need a cancel method. 

```python
# Set up a callback to a function
def after(self, time, function_to_call):
    time_in_secs = int(time * 1000)
    self.tk.after(time_in_secs, function_to_call)

# Schedule a function every n seconds
def every(self, time, function_to_call):
   time_in_secs = int(time * 1000)

    # Take their function but make it self-call again
    def func_with_self_callback():
        function_to_call()
        self.tk.after(time_in_secs, func_with_self_callback)

    # Register a callback to this function
    self.tk.after(time_in_secs, func_with_self_callback)
```

So my question is - is this a total disaster for a reason I haven't spotted? It feels like a trap...
cc @bennuttall

**Comment by Coal0** [comment]
This should work as expected. To implement a 'stoppable' thread-like version, maybe consider:

```python
class _Foo:
    def __init__(self):
        self._callback = []
        # Store all callable events

    def after(self, time, function_to_call):
        """Call `function_to_call` after `time` milliseconds."""
        self.tk.after(time, function_to_call)

    def every(self, time, function_to_call):
        """Call `function_to_call` every `time` milliseconds."""
        self._callback[function_to_call] = True
        def _call_wrapper():
            if self._callback[function_to_call]:
                function_to_call()
                self.tk.after(time, _call_wrapper)
            else:
                return
        self.tk.after(time, _call_wrapper)

    def cancel(self, function_to_call):
        """Cancel the scheduled `function_to_call`."""
        self._callback[function_to_call] = False
```

This is making the assumption that `every` calls will have unique `function_to_call` arguments. Some other things that caught my attention:

* `tk.after` expects milliseconds, so why are you converting `time` to seconds?
* Maybe rename `every` to `schedule` or `repeat`?

**Comment by lawsie** [comment]
Fixed #57
